File: resources/public/js/artwork.js
(function() {
  var Artwork;
  Artwork = (function() {
    function Artwork() {
      this.album = void 0;
      this.artist = void 0;
      $('#current_track').delegate('.album-cover', 'ajax:success', this.render);
      this.load();
      $('#current_track').bind('track.updated', this.load);
    }
    Artwork.prototype.load = function() {
      var $covers;
      $covers = $('.album-cover');
      return $covers.each(function() {
        var $cover;
        $cover = $(this);
        this.album = $cover.attr('data-album');
        this.artist = $cover.attr('data-artist');
        return $.get('http://ws.audioscrobbler.com/2.0/', {
          "method": "album.getInfo",
          "api_key": "809bf298f1f11c57fbb680b1befdf476",
          "album": this.album,
          "artist": this.artist
        }, function(data) {
          return $cover.trigger('ajax:success', [data]);
        }).error(function(data) {
          return $cover.trigger('ajax:success', [null]);
        });
      });
    };
    Artwork.prototype.render = function(e, info) {
      var img, size, src, wrap;
      size = $(this).attr('data-thumbnail');
      src = $(info).find("image[size=" + size + "]").text();
      if (src === '') {
        src = '/img/no_art_lrg.png';
      }
      wrap = $('<a />', {
        'href': '#'
      });
      img = $('<img>', {
        'class': 'thumbnail',
        'src': src
      });
      wrap.append(img);
      return $(this).append(wrap);
    };
    return Artwork;
  })();
  $(function() {
    return new Artwork;
  });
}).call(this);

File: src/jukebox_web/models/artwork.clj
(ns jukebox-web.models.artwork
  (:require [clj-http.client :as client]
            [ring.util.codec :as codec])
  (:use [clojure.data.json :only (read-str)]
        [clojure.java.io :only (reader)]))

(def default-image-path "/img/no_art_lrg.png")

(def default-images {:large default-image-path :extra-large default-image-path})

(def base-url "http://ws.audioscrobbler.com/2.0?method=album.getInfo&api_key=809bf298f1f11c57fbb680b1befdf476&format=json&autocorrect=1")

(defn url [album artist]
  (format "%s&album=%s&artist=%s" base-url (codec/url-encode album) (codec/url-encode artist)))

(defn image-for-size [images size]
  (get (first (filter #(and (= (:size %) size) (not (empty? (:#text %)))) images)) "#text"))

(defn transform [http-response]
  (let [json-response (read-str (:body http-response))
        images (get-in json-response ["album" "image"])]
    (if (empty? images)
      default-images
      {:large (image-for-size images "large")
       :extra-large (image-for-size images "mega")})))

(defn album-cover [album artist]
  (try
    (let [response (client/get (url album artist))]
      (if (= 200 (:status response))
        (transform response)
        default-images))
    (catch java.lang.Exception e (do
                                   (prn e)
                                   default-images))))

